src/main/java/com/microsoft/azure/vmagent/AzureVMAgentTemplate.java

```diff
@@ -1771,9 +1771,13 @@ public FormValidation doCheckSshConfig(
         @POST
         public FormValidation doCheckInitScript(
                 @QueryParameter String value,
-                @QueryParameter String agentLaunchMethod) {
+                @QueryParameter String agentLaunchMethod,
+                @QueryParameter String imageTopLevelType) {
+            if (Constants.IMAGE_TOP_LEVEL_BASIC.equals(imageTopLevelType)) {
+                return FormValidation.ok();
+            }
             if (StringUtils.isBlank(value)) {
-                return FormValidation.warningWithMarkup(Messages.Azure_GC_InitScript_Warn_Msg());
+                return FormValidation.okWithMarkup(Messages.Azure_GC_InitScript_Warn_Msg());
```

Comment:
used `ok` so-as not to worry people using built-in or that just don't need an init script